tempesta_fw/str.h

```diff
@@ -99,27 +99,32 @@
  * 		  string and the least significant byte for flags;
  */
 typedef struct {
-	void		*ptr;
 	struct sk_buff	*skb;
 	unsigned long	len;
 	unsigned char	eolen;
-	unsigned int	flags;
+
+	unsigned int	chunknum : 24;
+	unsigned int	flags : 8;
```

Comment:
I don't think it makes sense to mix the two different ways of handling the flags and the number of chunk. Does it really? `In my opinion, `__str_grow_tree()` should be fixed to work directly with the number of chunks.